### ISSUE TITLE
Thread Mentions

### ISSUE THREAD

**Issue by aboydnw** [NASA-IMPACT/nasa-apt#456]
## Acceptance Criteria

- [x] Given I am a contributor, reviewer, or curator, when I leave a comment on an ATBD then I am able to select one or multiple authors to receive the notification
- [x] Given I am a contributor, reviewer, or curator, when I leave a comment on an ATBD then the default is for that comment to be sent to All contributors, unless otherwise specified
- [x] Given I have left a comment on an ATBD, when the notification is sent out for that comment then only people specified in the comment receive the notification (this could be one, multiple, or all contributors)
- [ ] ~Tested to ensure notifications are working correctly (Brad)~

**Comment by aboydnw** [comment]
Potential solution:
Use existing comment UI and add a multi-select dropdown that includes all the collaborators (and curators?) for that report that is defaulted to an "All" option

Based on the selection from that dropdown, the comment notification is only sent to those users


![Image](https://user-images.githubusercontent.com/92742765/171512493-9b069fbd-7423-4230-9da9-5de8c1e87929.png)

**Comment by oliverroick** [comment]
@aboydnw @deborahUAH 

Started some front-end work on this ticket and want to clarify an assumption

The notification selection only applies to one comment. For responses to that comment to that comment the selection doesn't apply. That means Jane writes a comment and selects to notify Betty, when Betty replies to Jane's comment, the notification will be sent to all users unless Betty selects the people who should be notified. 

Is this correct? To me it feels like this is wrong. Should the selection apply to the whole comment thread?

**Comment by aboydnw** [comment]
Ah good catch @oliverroick. I agree, we should probably change that. My proposal would be that we use a similar UI on the reply, adding a dropbox like in this ticket. Would it be possible to have it default to whatever the selection was made from the original post? As opposed to defaulting to All. Then the user could change it if they need, but also easily not do anything and have the same selection.

What do you think?

**Comment by oliverroick** [comment]
Yes, we can make that work. :+1:

**Comment by aboydnw** [comment]
1. Included in the list of recipients for a comment would be "Curator". If it is selected, then all users with Curator user group receive the notification
2. For all curators, when they leave a comment on an ATBD, the comment is identified as coming from a curator

One idea to identify as curator is to append " (Curator)" to the end of the name.

@oliverroick @naomatheus Recording a couple bullets based on the conversation during today's tagup. Oliver, Matt said he would sync with you on these.

We can do this under this ticket or under a new one.

**Comment by aboydnw** [comment]
Adding some more context for the comment above...the main reason to include the curators somehow here is because the curators are playing a much more active role than previously assumed. They will have to have some back and forth with the authors, in particular around journal publication and progress of the document. One idea is to enable users to have this back and forth via comments.

**Comment by oliverroick** [comment]
Using comments to discuss journal publication and general progress doesn't sound like an appropriate solution to me. Comments are linked to sections in the document. Discussions around journal publication are generic, not linked to a specific section. Where would these type of conversations happen? Under any section? The generic section? In either case, they are going to distract from the comments around editing individual sections.

The proper response would probably be a discussions section for each document away from the editing process.

**Comment by aboydnw** [comment]
@oliverroick I think the idea was that this discussion would happen on the Journal Details sections, on the last page of the document. Although it is a good point that comments are not a good spot for general questions about progress of the document. 
![image](https://user-images.githubusercontent.com/92742765/182190061-cada8074-4c17-4b1c-8f8f-abb5eaec18a4.png)

**Comment by oliverroick** [comment]
@aboydnw (cc: @wrynearson) 

I've worked on the curator comments today. I've got a solution for notifying all curators, but want to run a design decision by you. Adding the curator label can be done two ways: Just adding a label in brackets or adding a label component, which is more in line with the rest of the applications. 


![Image](https://user-images.githubusercontent.com/159510/184875779-d9e2ef1c-d6d7-427b-8e39-b2e77668b88f.png)
![Image](https://user-images.githubusercontent.com/159510/184875778-5e6414f4-5445-4c86-aa5f-e915346536f8.png)

Both solution aren't ideal, as you can see either the curator label or the user name are truncated. We can make the comment panel a bit wider to solve this, the only flaw is that it would cover more of the editor:

![Image](https://user-images.githubusercontent.com/159510/184876023-8737d69a-9775-4b82-a340-c186204cb644.png)

So two things to decide:

1. The curator label as text in brackets or with the label component?
2. Ok to make the comment panel wider

**Comment by aboydnw** [comment]
@oliverroick Since I am in NYC with Ricardo and Daniel, I got their quick opinions. They thought it was a good direction. A couple thoughts Ricardo had:
1. Maybe we can use an icon instead of the tile? (I personally liked the tile over the parentheses, maybe icon would be even better) That way, it could take up less space. He referenced the lead author icon (in the collaborators dropdown) as an example of where we are using icons already.
2. He agreed with the idea of stretching the panel to make more room. How much we stretch this is maybe dependent on whether we use a tile or icon. 

Overall I like the tile and stretching the panel. Maybe Z Creative would be interested in helping out with some of these design questions? Not sure how that might work for billing, though. cc @wrynearson

**Comment by oliverroick** [comment]
Icon would look like this:

![Screenshot 2022-08-17 at 15 10 55](https://user-images.githubusercontent.com/159510/185142729-bbf8894f-2f4a-4cc9-93ef-5ce655108a19.png)

Personally, I find that an icon is the least favourable option because it conveys no information.

**Comment by deborahUAH** [comment]
I personally like the first image 
<img width="167" alt="image" src="https://user-images.githubusercontent.com/88099931/185184330-6d374a49-0ce7-46c5-b7d2-f42dc8461e9f.png">
best since it communicates the clearest.  Name isn't as important as the user knowing it is the APT curator communicating
stretching a bit is good.

**Comment by aboydnw** [comment]
Okay, let's go with the tile and stretching the panel like you suggested @oliverroick

**Comment by wrynearson** [comment]
This is ready to be deployed

**Comment by deborahUAH** [comment]
@naomatheus @wrynearson  what is status of this ticket?  It is ready but was never deployed?  I have a user (today 11/21) that got an error when leaving a comment.  Is there a holdup to this push?  Thanks for an update.

**Comment by wrynearson** [comment]
Hi @deborahUAH, thanks for flagging this. The feature has been merged into production with (https://github.com/NASA-IMPACT/nasa-apt-frontend/pull/397). 

However, I just tested it on production and also received an error in posting a comment.

@naomatheus, here's the console log I received:
```
Access to XMLHttpRequest at 'https://8864dn8eqh.execute-api.us-west-2.amazonaws.com/v2/threads' from origin 'https://www.earthdata.nasa.gov' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```

We had a similar error with SNWG. @naomatheus you'll need to add the URL to the whitelist in the API Gateway settings in MCP.

**Comment by wrynearson** [comment]
@naomatheus, can you confirm that this was merged into production? If so, we do have an issue with the production thread mentions.

**Comment by bwbaker1** [comment]
@wrynearson @naomatheus @deborahUAH, I tested the notification aspect of this feature and it does not seem to work.  The steps I tried: (1) add two authors to a document - contributor@apt.com and bwb0013@uah.edu (profile name Nevada!11271984) (2) while logged in as contributor@apt.com, I left a comment and selected to notify Nevada!11271984 (3) and looked for email in my bwb0013@uah.edu inbox.  I did not receive an email notification.



![Image](https://user-images.githubusercontent.com/56319064/214594918-994b688d-57c5-40ee-adba-5c16de978255.png)

**Comment by wrynearson** [comment]
Thanks @bwbaker1, sad to hear it's not working. I'm sure you already checked for this, but could you see if it went to your spam folder? That happened with SNWG sometimes

**Comment by bwbaker1** [comment]
@wrynearson I double-checked all of my folders again, and unfortunately I didn't receive anything.

**Comment by wrynearson** [comment]
Ok thanks for the update. I'll ask Iksha to make me a curator and then we can test this internally.

**Comment by wrynearson** [comment]
I tested this on staging just now and also didn't receive notifications. However, I think we should wait to close this ticket (and address the problem) until we are working on all of the other email notifications, because technically, that AC relates more to that body of work.

**Comment by bwbaker1** [comment]
@wrynearson @deborahUAH @cw0045 @ashleyriddle92 This has been tested and is ready for Production.
